docs/dev-environments/ethereum/besu/startup.sh

```diff
@@ -32,10 +33,23 @@ echo ""
 echo "========================================================="
 echo "STEP 3 :: Initialize truffle project and deploy contracts"
 echo "========================================================="
-docker-compose -f docker-compose-truffle.yaml up \
-|| error_exit "Failed to initialize truffle or deploy contracts"
+
+# Check if TCF_HOME is defined and current working directory
+# is a subtree of TCF_HOME
+if [[ -z "$TCF_HOME" ]] || [[ `pwd` != "$TCF_HOME"* ]];
+then
+    error_exit "TCF_HOME environment variable not defined appropriately. Please check."
+fi
+
+# Verify if contract deployments are successful else log error
+docker-compose -f docker-compose-truffle.yaml up | grep -e "^" -e "Total deployments:   3"
```

Comment:
Check if the deployment is successful by checking the command that's being executed as part of docker-compose